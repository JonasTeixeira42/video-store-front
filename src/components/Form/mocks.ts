export default [
  {
    label: 'Email',
    labelFor: 'Email',
    placeholder: 'Email',
  },

  {
    label: 'Password',
    labelFor: 'Password',
    placeholder: 'Password',
  },
];
