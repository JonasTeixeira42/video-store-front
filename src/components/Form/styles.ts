import styled, { css } from 'styled-components';

import * as ButtonStyles from 'components/Button/styles';
import * as TextFieldStyles from 'components/TextField/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${TextFieldStyles.Wrapper} {
      margin: ${theme.spacings.xxsmall} 0;
    }

    ${ButtonStyles.Wrapper} {
      margin-left: auto;
      margin-top: ${theme.spacings.medium};
    }
  `}
`;

export const FormWrapper = styled.form``;
