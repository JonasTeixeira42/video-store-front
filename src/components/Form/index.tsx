import React, { FormEvent } from 'react';
import Button from 'components/Button';
import TextField, { TextFieldProps } from 'components/TextField';

import * as S from './styles';

export type FormProps = {
  children: React.ReactNode;
  buttonLabel: string;
  buttonIcon: React.ReactNode;
  onSubmit?: (event: FormEvent<HTMLFormElement>) => void;
};

const Form = ({ children, buttonLabel, buttonIcon, onSubmit }: FormProps) => {
  return (
    <S.Wrapper>
      <S.FormWrapper onSubmit={onSubmit}>
        {children}

        <Button size="small" type="submit" icon={buttonIcon}>
          {buttonLabel}
        </Button>
      </S.FormWrapper>
    </S.Wrapper>
  );
};

export default Form;
