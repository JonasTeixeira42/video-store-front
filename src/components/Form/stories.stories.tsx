import { Story, Meta } from '@storybook/react/types-6-0';
import { AddBox } from '@styled-icons/material-outlined/';
import TextField from 'components/TextField';

import Form, { FormProps } from '.';

export default {
  title: 'Form',
  component: Form,
  args: {
    buttonLabel: 'Cadastrar',
    buttonIcon: <AddBox size={24} />,
  },
  argTypes: {
    buttonIcon: {
      type: '',
    },
  },
} as Meta;

export const Default: Story<FormProps> = (args) => (
  <div style={{ maxWidth: '42rem', margin: 'auto' }}>
    <Form {...args}>
      <TextField label="email" placeholder="email" name="email" />
      <TextField label="password" placeholder="password" name="password" />
    </Form>
  </div>
);
