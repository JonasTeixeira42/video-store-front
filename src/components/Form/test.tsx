import { screen } from '@testing-library/react';
import { AddBox } from '@styled-icons/material-outlined/';

import Form from '.';
import { renderWithTheme } from 'utils/tests/helpers';

import inputs from './mocks';

describe('<Form />', () => {
  it('should render the form', () => {
    const { container } = renderWithTheme(
      <Form
        inputs={inputs}
        buttonLabel="Cadastrar"
        buttonIcon={<AddBox size={24} />}
      />,
    );

    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: /cadastrar/i }),
    ).toBeInTheDocument();

    expect(container.parentElement).toMatchSnapshot();
  });
});
