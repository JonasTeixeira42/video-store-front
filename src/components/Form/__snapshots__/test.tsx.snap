// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render the form 1`] = `
.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 600;
  font-size: 1.6rem;
  border-radius: 0.8rem;
  border: 0;
  outline: 0;
  background: #04d361;
  color: #FAFAFA;
  cursor: pointer;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  background: #03a14a;
}

.c5:hover {
  background: #03a14a;
}

.c5 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1.6rem;
  background: #04d361;
  border-radius: 0 0.8rem 0.8rem 0;
  margin: 0 auto;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
}

.c5:hover {
  background: #026f33;
}

.c5:hover > div {
  background: #03a14a;
}

.c5 p,
.c5 > div {
  padding: 1.6rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #EAEAEA;
  border-radius: 0.2rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c2:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c3 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c1 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
}

.c0 .{
  margin: 0.8rem 0;
}

.c0 .c4 {
  margin-left: auto;
  margin-top: 3.2rem;
}

<body>
  <div>
    <div
      class="c0"
    >
      <form
        class=""
      >
        <div
          class=""
        >
          <label
            class="c1"
            for=""
          >
            email
          </label>
          <div
            class="c2"
          >
            <input
              class="c3"
              name="email"
              placeholder="email"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=""
        >
          <label
            class="c1"
            for=""
          >
            password
          </label>
          <div
            class="c2"
          >
            <input
              class="c3"
              name="password"
              placeholder="password"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="c4 c5"
          color="white"
          type="submit"
        >
          <p>
            Cadastrar
          </p>
          <div>
            <svg
              aria-hidden="true"
              class="c6"
              fill="currentColor"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zm-8-2h2v-4h4v-2h-4V7h-2v4H7v2h4z"
              />
            </svg>
          </div>
        </button>
      </form>
    </div>
  </div>
</body>
`;
